src/fairseq2/models/wav2vec2/frontend.py

```diff
@@ -161,9 +161,11 @@ def extract_features(
         if self.feature_extractor is not None:
             seqs, padding_mask = self.feature_extractor(seqs, padding_mask)
 
+        feature_penalty = seqs.float().pow(2).mean()
```

Comment:
I suggest:

```python
raw_features = seqs

seqs = self.post_extract_layer_norm(seqs)

return seqs, padding_mask, raw_features
```

and computing the `float().pow(2).mean()` (or better yet use `torch.linalg.vector_norm()`) only during loss calculation if the penalty weight is greater than zero.

Reply:
+1 to returning `raw_features` (and not `feature_penalty`). @cbalioglu do you want to consider returning an object here (instead of a tuple of tensors) ?

Reply:
Yes, I think it makes sense to convert this to a proper dataclass instead of a tuple at this point. Once we land these changes, let's make sure that we reflect them in your repo. We are pretty close to v0.3 release, it would be good timing.